test/TestProjects/Accessibility-LowLevel/Generated/AccessibilityClient.cs

```diff
@@ -43,30 +41,91 @@ public AccessibilityClient(AzureKeyCredential credential, Uri endpoint = null, A
+                {
+                    switch (message.Response.Status)
+                    {
+                        case 200:
+                            return message.Response;
+                        default:
+                            throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
+                    }
+                }
+                else
+                {
+                    return message.Response;
+                }
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
         }
 
         /// <param name="requestBody"> The request body. </param>
-        /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual Response Operation(RequestContent requestBody, CancellationToken cancellationToken = default)
+        /// <param name="requestOptions"> The request options. </param>
+        public virtual Response Operation(RequestContent requestBody, RequestOptions requestOptions = null)
         {
-            Request req = CreateOperationRequest(requestBody);
-            return Pipeline.SendRequest(req, cancellationToken);
+            requestOptions ??= new RequestOptions();
+            HttpMessage message = CreateOperationRequest(requestBody, requestOptions);
+            if (requestOptions.PerCallPolicy != null)
+            {
+                message.SetProperty("RequestOptionsPerCallPolicyCallback", requestOptions.PerCallPolicy);
+            }
+            using var scope = _clientDiagnostics.CreateScope("AccessibilityClient.Operation");
+            scope.Start();
+            try
+            {
+                Pipeline.Send(message, requestOptions.CancellationToken);
+                if (requestOptions.StatusOption == ResponseStatusOption.Default)
+                {
+                    switch (message.Response.Status)
+                    {
+                        case 200:
+                            return message.Response;
+                        default:
+                            throw _clientDiagnostics.CreateRequestFailedException(message.Response);
+                    }
+                }
+                else
+                {
+                    return message.Response;
+                }
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
         }
 
         /// <summary> Create Request for <see cref="Operation"/> and <see cref="OperationAsync"/> operations. </summary>
         /// <param name="requestBody"> The request body. </param>
-        private Request CreateOperationRequest(RequestContent requestBody)
+        /// <param name="requestOptions"> The request options. </param>
+        private HttpMessage CreateOperationRequest(RequestContent requestBody, RequestOptions requestOptions = null)
```

Comment:
The new parameter is never used.

Reply:
That's because we're injecting it here I believe:

https://github.com/Azure/autorest.csharp/pull/1219#discussion_r628427777

Reply:
Hmm, I'm not sure if we can fix this without pull RequestOptions injection out of the model (or special casing it in 2 places)

Reply:
Filed as https://github.com/Azure/autorest.csharp/issues/1222